### PR TITLE
filter audio on sender side, in attempt to fix webrtc bug

### PR DESCRIPTION
**Pull request by kfinn**
best guess at fixing https://github.com/kfinn/casual_rails/issues/18

let's test it together though @samandmoore

### CODE REVIEW

File: app/javascript/channels/consumer.ts [hunk 1]
```diff
@@ -22,23 +22,19 @@ export default {
                 { channel, id },
                 {
                     connected() {
-                        console.log(`connected to ${channel} ${id}`)
```
Comment (kfinn) [inline app/javascript/channels/consumer.ts:25]: oh, I felt like all these logs had very little value to me anymore, so ✂️ 

but we can bring them back if you still liked them

File: app/javascript/components/useFilteredLocalStream.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    const membershipPairIsWhisperTarget = useRecoilValue(membershipPairIsWhisperTargetState(membershipPairId))
+    const whisperTargetMembershipPairIdExists = useRecoilValue(whisperTargetMembershipPairIdExistsState)
+    const muted = useRecoilValue(mutedState)
+
+    const groupId = useRecoilValue(groupIdState)
+
+    const [status, setStatus] = useState<StreamStatus>(StreamStatus.ENABLED)
+
+    useEffect(() => {
+        if (
```
Comment (kfinn) [inline app/javascript/components/useFilteredLocalStream.ts:40]: it was hard to think about this logic backwards

**Comment by kfinn** [comment]
spotted a surprising behavior in group creation that I fixed, too:

surprising behavior:
- all users are muted
- user 1 whispers user 2
- user 2 whispers user 1
- users 1 and 2 can hear each other
- user 1 creates group with user 2
- users 1 and 2 receive the group creation message, and stop whispering each other
- all users remain muted
- users 1 and 2 cannot hear each other

fix:
- all users are muted
- user 1 whispers user 2
- user 2 whispers user 1
- users 1 and 2 can hear each other
- user 1 creates group with user 2
- users 1 and 2 receive the group creation message, and stop whispering each other **and stop being muted**
- users 1 and 2 are unmuted, no other user mute status is changed
- users 1 and 2 can hear each other
- other users her quiet audio from users 1 and 2

File: app/models/membership_pair_group_creation.rb [hunk 3]
```diff
@@ -10,17 +10,12 @@ def save!
...
+                Group.create!(
+                    memberships: (room.memberships - membership_pair.memberships)
+                )
             end
 
-            group.save!
-            membership_pair.memberships.each do |membership|
-                membership.update! group: group
-            end
+            Group.create! memberships: membership_pair.memberships
```
Comment (kfinn) [inline app/models/membership_pair_group_creation.rb:18]: I was rereading this trying to see if I had broken group creation, and felt like this was more descriptive/cuter

File: app/javascript/components/useFilteredLocalStream.ts (hunk 2 above)
Comment (samandmoore) [reply app/javascript/components/useFilteredLocalStream.ts:40]: Looks right to me! But it's definitely hard to think about

**Comment by samandmoore** [comment]
Looks good! Let's test it out today

**Pull request merged**